repo.go

```diff
@@ -348,17 +324,13 @@ func (r *Repo) GenKeyWithExpires(keyRole string, expires time.Time) (keyids []st
 }
 
 func (r *Repo) AddPrivateKey(role string, signer keys.Signer) error {
+	// Not compatible with delegated roles.
+
 	return r.AddPrivateKeyWithExpires(role, signer, data.DefaultExpires(role))
 }
 
 func (r *Repo) AddPrivateKeyWithExpires(keyRole string, signer keys.Signer, expires time.Time) error {
-	if !verify.ValidRole(keyRole) {
-		return ErrInvalidRole{keyRole}
-	}
-
-	if !validExpires(expires) {
-		return ErrInvalidExpires{expires}
-	}
```

Comment:
I see this is moved to `AddVerificationKeyWithExpiration` and so is it okay to verify the role/expiration **after** we do `SaveSigner()`?

Reply:
I don't think there is a security impact here, but in https://github.com/theupdateframework/go-tuf/pull/175/commits/00996cacd27ad5591d35e54ce50d93a36708adaa I reordered these calls so the verification key is added (with the given expiration date) before the signer is added.

Reply:
Actually this commit made lots of tests fail. Working on figuring out why.

Reply:
Addressed in https://github.com/theupdateframework/go-tuf/pull/175/commits/a983813a0fb5f631dee3352704b19898e2d00d7a: "Revert 00996cacd, add back arg validation to AddPrivateKeyWithExpires, and add comment about func call order"